services/devicefinder/devicefinder/discovery/configmap.go

```diff
@@ -0,0 +1,102 @@
+package discovery
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/red-hat-storage/ocs-operator/v4/services/devicefinder/types"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+const (
+	configMapNamePrefix = "devicefinder-result-"
+)
+
+// updateConfigMap creates or updates the ConfigMap with discovered devices
+func (discovery *DeviceDiscovery) updateConfigMap() error {
+	nodeName := os.Getenv("MY_NODE_NAME")
+	namespace := os.Getenv("WATCH_NAMESPACE")
+
+	if nodeName == "" || namespace == "" {
+		return fmt.Errorf("missing required environment variables: MY_NODE_NAME or WATCH_NAMESPACE")
+	}
+
+	// Create Kubernetes client
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return fmt.Errorf("failed to get in-cluster config: %w", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create kubernetes client: %w", err)
+	}
```

Comment:
we should get the env and create the client only one and make this as member of DeviceDiscovery struct and reuse it